_source/_posts/2021-12-14-antipatterns-secrets-java.adoc

```diff
@@ -0,0 +1,147 @@
+---
+layout: blog_post
+title: "Five Anti-Patterns with Secrets in Java"
+author: brian-demers
+by: advocate
+communities: [security,java]
+description: "Learn how how to better manage and protect your passwords and API keys!"
+tags: [java, security, secrets, cryptography, tips]
+tweets:
+- "When was the last time you checked a password into source control? 🤦 Here are a few tips to avoid that and more‼️"
+- "API keys, passwords, and secrets. Learn a few common anti-patterns to avoid ⚠️"
+- "Secrets, passwords, and API keys oh my‼️ Make sure you are handling them correctly 🧑‍💻"
+image:
+type: awareness
+---
+
+:toc: macro
+:page-liquid:
+:experimental:
+
+Most applications require some sort of secret or password to enable access: database connection info, API keys, OAuth client secrets, and JWT keys, to list some examples. Dealing with secrets in your projects is always a chore, and it's often done wrong. In this post, I'll describe five common problems, which you can think of as *anti-patterns* (the opposite of a *best practice*) and I'll offer suggestions to help you avoid these issues.
+
+toc::[]
+
+== 1. Don't check your secrets into source control
```

Comment:
If these are anti-patterns, shouldn't this be more like "Check your secrets into source control"?

Reply:
```suggestion
== 1. Checking secrets into source control
```
to Matt's point.